ui/lib/kv/addon/components/page/secret/edit.js

```diff
@@ -44,6 +52,22 @@ export default class KvSecretEdit extends Component {
     return false;
   }
 
+  get diffDelta() {
+    const oldData = JSON.parse(this.originalSecret);
```

Comment:
I might be reading this wrong - but if we're stringify-ing above and parsing here - could we skip the JSON-ing and just save the original secret directly as `this.args.secret.secretData`?

Reply:
No, since the data is changing on the passed arg (we refer to it as `newData`) we have to make a copy on init. Storing the original as a JSON.stringify copy is basically copying it without keeping to reference to the arg. Another option is to make a spread copy like `this.originalData = {...this.args.secret.secretData }`, would that be more obvious why?

Reply:
Ah yeah - that totally makes sense. Javascript 😅  Spreading is more obvious to me but that's personal preference - I think it's totally fine as is!